libcudacxx/include/cuda/std/__bit/reference.h

```diff
@@ -77,68 +87,84 @@ class __bit_reference
     return *this;
   }
 
-  _LIBCUDACXX_INLINE_VISIBILITY __bit_reference& operator=(const __bit_reference& __x) noexcept
+#if _CCCL_STD_VER >= 2023
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY constexpr const __bit_reference&
+  operator=(bool __x) const noexcept
+  {
+    if (__x)
+    {
+      *__seg_ |= __mask_;
+    }
+    else
+    {
+      *__seg_ &= ~__mask_;
+    }
+    return *this;
+  }
+#endif // C++23+
+
+  _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 __bit_reference&
+  operator=(const __bit_reference& __x) noexcept
   {
     return operator=(static_cast<bool>(__x));
   }
```

Comment:
Is there any reason we need this actual implementation or can we just drop the definition of the SMF alltogether?

Reply:
Yes, this is a proxy type.